openml-h2o/src/main/java/com/feedzai/openml/h2o/algos/AbstractH2OAlgoUtils.java

```diff
@@ -0,0 +1,169 @@
+     * @param randomSeed The source of randomness.
+     * @return The modified version of the given {@code h2oParams}.
+     */
+    protected abstract T parseSpecificParams(final T h2oParams, final Map<String, String> params, final long randomSeed);
+
+    /**
+     * Returns an empty representation of the algorithm specific parameters.
+     *
+     * @return An empty representation of the algorithm specific parameters.
+     */
+    protected abstract T getEmptyParams();
+
+    /**
+     * Template method to parse H2O algorithm params.
+     *
+     * @param trainingFrame The dataset to be used.
+     * @param params        The raw training params.
+     * @param randomSeed    The source of randomness.
+     * @param datasetSchema The dataset schema.
+     * @return The modified version of the given {@code h2oParams}.
+     */
+    protected final T parseParams(final Frame trainingFrame,
+                               final Map<String, String> params,
+                               final long randomSeed,
+                               final DatasetSchema datasetSchema) {
+        final T baseParams = commonParams(trainingFrame, datasetSchema);
+        return parseSpecificParams(baseParams, params, randomSeed);
+    }
+
+    /**
+     * Auxiliary method to setup the common training params to all algorithms/models.
+     *
+     * @param trainingFrame The dataset to be used.
+     * @param datasetSchema The dataset schema.
+     * @return A modified version of the provided params object.
+     */
+    protected abstract T commonParams(Frame trainingFrame, DatasetSchema datasetSchema);
+
+    /**
+     * Abstract method that creates and returns the Model that this class refers to.
+     * @param h2oParams concrete implementation of {@link ModelParametersSchemaV3 algorithm params}.
+     * @return concrete implementation of {@link ModelBuilder H20 Model Builder} created from the h2oParams.
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    public abstract M getModel(T h2oParams);
+
+    /**
+     * Validates the h20 model parameters by building a model and validating it.
+     * @param paramsToValidate H20 parameters to validate.
+     * @param randomSeed       The source of randomness.
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    public void validateParams(Map<String, String> paramsToValidate, final long randomSeed){
+        final T baseParams = getEmptyParams();
+        final T updatedParams = parseSpecificParams(baseParams, paramsToValidate, randomSeed);
+        M modelBuilder = getModel(updatedParams);
+        validateModel(modelBuilder);
+    }
+
+    /**
+     * Trains the H20 model.
+     * @param trainingFrame The dataset to be used.
+     * @param params        The raw training params.
+     * @param randomSeed    The source of randomness.
+     * @param schema        The dataset schema.
+     * @return Job resulting from he model training.
+     *
+     * @since @@@feedzai.next.release@@@
```

Comment:
AFAIK, openml project does not update `@since @@@feedzai.next.release@@@` automatically,
So either you add since with next release version, or update it manually when doing release

This applies to all following since tag

Reply:
In the POM it has no version set, it appears to be changed during release. Should I check the last couple of releases and infer what the next one will be ?

Reply:
the version on pom is updated, but the since tag is not

Reply:
we are using jgitver for versioning management

the version are specified using git tags
e.g. create a tag with 19.1.1 and it will release a new version of the repo with that version